torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2300,22 +2320,21 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::endCoalescing(OpType optype) {
     work->stashed_for_allocator_safety_ =
         std::make_shared<std::vector<at::Tensor>>();
   }
-  c10::cuda::CaptureStatus capture_status =
-      c10::cuda::currentStreamCaptureStatusMayInitCtx();
 
   // Notify graphs before we check the capture status preemptively
   at::cuda::CUDAGraph::inc_pending_event_queries();
 
-  if ((coalescing_state_ & CoalColl) &&
-      capture_status == c10::cuda::CaptureStatus::None) {
+  if (enqueue) {
     workEnqueue(work);
-    // TODO: it seems we never enqueue work for single send/recv or batch P2P,
-    // see the `pointToPoint` function. This should be fixed. Otherwise, we risk
-    // not being able to abort hanged P2P ops.
   } else {
     at::cuda::CUDAGraph::dec_pending_event_queries();
   }
 
+  // Bump collective counter after creating the work-
+  // this way all the flight-recorder events inside the coalesing group share
+  // the seq_ with the coalsed op and the work that goes with it
+  seq_++;
```

Comment:
Should we bump the seq_ only if enqueue is true? my understanding is that there is a one to one mapping of 
seq_ -> enqueued work?

Reply:
well, see my latest version and then lets talk. I wonder if you will agree with my new approach or not.  (didn't see your comment before i pushed.)